utils/findSelectedItem.ts

```diff
@@ -1,15 +1,27 @@
+import { redirect } from '@/navigation';
+
 import { replaceDashWithSpace } from './string';
 
 export const findSelectedItem = <T extends { name: string; engName?: string }>(
   items: T[],
-  dashedItemName?: string,
+  selectedItemName?: string,
+  pathname?: string,
 ) => {
   const defaultItem = items[0];
 
-  if (dashedItemName === undefined) return defaultItem;
+  if (selectedItemName === undefined) return defaultItem; // 쿼리 없이 기본 주소일 경우
+
+  const selectedName = replaceDashWithSpace(selectedItemName);
+  const item = items.find(
+    (item) =>
+      replaceDashWithSpace(item.name) === selectedName ||
+      replaceDashWithSpace(item.engName ?? '') === selectedName,
```

Comment:
빈 문자열에 `replaceDashWithSpace`를 할 필요는 없으니까 바깥으로 빼는건 어때?
```suggestion
      (item.engName && replaceDashWithSpace(item.engName) === selectedName),
```